WalletWasabi/WabiSabi/Backend/Rounds/CoinJoinStorage/CoinJoinIdStore.cs

```diff
@@ -0,0 +1,63 @@
+using NBitcoin;
+using System.IO;
+using System.Linq;
+using WalletWasabi.Logging;
+
+namespace WalletWasabi.WabiSabi.Backend.Rounds.CoinJoinStorage;
+
+public class CoinJoinIdStore : ICoinJoinIdStore
+{
+	private InMemoryCoinJoinIdStore InMemoryCoinJoinIdStore { get; set; }
+	public string CoinJoinIdStoreFilePath { get; set; }
+	private object FileWriteLock { get; set; } = new();
+
+	public CoinJoinIdStore() : this(string.Empty)
+	{
+	}
+
+	public CoinJoinIdStore(string coinJoinIdStoreFilePath)
+	{
+		CoinJoinIdStoreFilePath = coinJoinIdStoreFilePath;
+		InMemoryCoinJoinIdStore = InMemoryCoinJoinIdStore.LoadFromFile(coinJoinIdStoreFilePath);
+	}
+
+	public void Append(uint256 id)
+	{
+		InMemoryCoinJoinIdStore.Add(id);
+		try
+		{
+			lock (FileWriteLock)
+			{
+				File.AppendAllLines(CoinJoinIdStoreFilePath, new[] { id.ToString() });
+			}
+		}
+		catch (Exception ex)
+		{
+			Logger.LogError($"Could not write to file {CoinJoinIdStoreFilePath}.", ex);
+		}
+	}
+
+	public bool Contains(uint256 id)
+	{
+		return InMemoryCoinJoinIdStore.Contains(id);
+	}
+
+	public static void ImportWW1CoinJoinsToWW2(string ww1CoinJoinsFilePath, string coinJoinIdStoreFilePath)
```

Comment:
I have an idea. This could be a static Create function that returns a CoinJoinIdStore object. 

`CoinJoinIdStore CoinJoinIdStore.Create( string coinJoinIdStoreFilePath, string ww1CoinJoinsFilePath,)`
The method should open both files, create IEnumerable<uint256> hashes without duplicates. If there was something new then write it into coinJoinIdStoreFilePath otherwise do nothing and then call CoinJoinIdStore ctor. 

The constructor could look like this:
public CoinJoinIdStore(IEnumerable<uint256> coinJoinHashes)



`InMemoryCoinJoinIdStore.LoadFromFile` can be deleted.